homeassistant/helpers/selector.py

```diff
@@ -207,8 +206,11 @@ class NumberSelector(Selector):
         has_min_max_if_slider,
     )
 
-    def __call__(self, data: Any) -> float:
+    def __call__(self, data: Any) -> float | None:
         """Validate the passed selection."""
+        if data is None:
```

Comment:
Why is `None` allowed as a valid number? If the person using this selector wants to allow `None`, they can mark it as optional in the schema.

Reply:
You're right, I've reverted that change.